presto-main/src/main/java/com/facebook/presto/translator/AbstractExpressionTranslator.java

```diff
@@ -0,0 +1,129 @@
+package com.facebook.presto.translator;
+
+import com.facebook.presto.spi.relation.CallExpression;
+import com.facebook.presto.spi.relation.ConstantExpression;
+import com.facebook.presto.spi.relation.InputReferenceExpression;
+import com.facebook.presto.spi.relation.LambdaDefinitionExpression;
+import com.facebook.presto.spi.relation.SpecialFormExpression;
+import com.facebook.presto.spi.relation.VariableReferenceExpression;
+import com.facebook.presto.spi.translator.ExpressionTranslator;
+import com.facebook.presto.spi.translator.TargetExpression;
+import com.facebook.presto.translator.registry.ConstantTypeRegistry;
+import com.facebook.presto.translator.registry.FunctionRegistry;
+
+import java.util.Map;
+
+public abstract class AbstractExpressionTranslator<T>
```

Comment:
User will extend this class when writing their translator. It will be populated with default implementations.